### PR TITLE
src: Fix up WebSocketShard errors

### LINKED ISSUE #3641: crash on WebsocketShard.onOpen
**Please describe the problem you are having in as much detail as possible:**
Seemingly at random, the bot crashed and i checked the error log and saw [this](https://cdn.discordapp.com/attachments/223867697312694272/658777384249589790/unknown.png), no idea how it happened, (im bad at describing so yell at me if i need to provide more info)

[WebsocketShard.js:252](https://github.com/discordjs/discord.js/blob/master/src/client/websocket/WebSocketShard.js#L252)
**Include a reproducible code sample here, if possible:**
dont have a reproduicble code sample

**Further details:**
- discord.js version: 12.0.0-dev
- Node.js version: v12.14.0
- Operating system: Debian 9
- Priority this issue should have – please be realistic and elaborate if possible: maybe between moderate and high

- [x] I have also tested the issue on latest master, commit hash: 710101c5805935d7804fc8e53fa130541f248c54

### LINKED ISSUE #3706: WebSocket onmessage connection is null
**Please describe the problem you are having in as much detail as possible:**
I don't know when, but at some time my bot give me this error:
```
UnhandledPromiseRejectionWarning: TypeError: Cannot set property 'onmessage' of null
    at WebSocketShard._cleanupConnection (C:\Projects\Tekato\node_modules\discord.js\src\client\websocket\WebSocketShard.js:718:31)
    at WebSocketShard.<anonymous> (C:\Projects\Tekato\node_modules\discord.js\src\client\websocket\WebSocketManager.js:248:15)
    at WebSocketShard.emit (events.js:210:5)
    at WebSocketShard.destroy (C:\Projects\Tekato\node_modules\discord.js\src\client\websocket\WebSocketShard.js:692:12)
    at WebSocketShard._send (C:\Projects\Tekato\node_modules\discord.js\src\client\websocket\WebSocketShard.js:640:12)
    at WebSocketShard.processQueue (C:\Projects\Tekato\node_modules\discord.js\src\client\websocket\WebSocketShard.js:666:12)
    at WebSocketShard.send (C:\Projects\Tekato\node_modules\discord.js\src\client\websocket\WebSocketShard.js:628:10)
    at WebSocketManager.broadcast (C:\Projects\Tekato\node_modules\discord.js\src\client\websocket\WebSocketManager.js:333:53)
    at ClientPresence.set (C:\Projects\Tekato\node_modules\discord.js\src\structures\ClientPresence.js:21:22)
```

This error appeared when i updated discord.js (i was already in master branch)

- discord.js version: master
- Node.js version: 12.14.0
- Operating system: Windows 10 LTSB
- WebSocket version: 7.2.1
- Priority this issue should have: High priority because i can't do anything in my bot.

- [X] I have also tested the issue on latest master, commit hash: 69c79a4136619e4dcacfffb935aa9bd60aa2081c

### PR DESCRIPTION
**Pull request by vladfrangu**
**Please describe the changes this PR makes and why it should be merged:**

In theory, this change should fix #3706 and should fix #3641! @xKesvaL, @NotSugden and @BannerBomb, please try this PR (since you guys reported the issue)

For anyone else reading this, if you've encountered any of the listed issues, please install this PR and reply here if it works or not! 

### How to install?

`npm i vladfrangu/discOwOd.js#fix-internal-sharding-again` or `yarn add vladfrangu/discOwOd.js#fix-internal-sharding-again`

### Changelog

The only notable thing is that `WebSocketShard#destroy` now takes an object of options and adds a few more options.

**Status**
- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**  
- [x] This PR changes the library's interface (methods or parameters added)
  - [x] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

File: src/client/websocket/WebSocketShard.js [hunk 1]
```diff
@@ -214,13 +214,12 @@ class WebSocketShard extends EventEmitter {
...
-        this.debug(`A connection was found. Cleaning up before continuing.
+        this.debug(`A connection object was found. Cleaning up before continuing.
     State: ${CONNECTION_STATE[this.connection.readyState]}`);
-        this._cleanupConnection();
-        this.connection.close(1000);
+        this.destroy({ emit: false });
       }
 
-      const wsQuery = { v: client.options.ws.version };
+      const wsQuery = { v: client.options.ws.version, vladdyUUID: Date.now().toString(36) };
```
Comment (amishshah) [inline src/client/websocket/WebSocketShard.js]: `vladdyUUID`?

Comment (vladfrangu) [reply src/client/websocket/WebSocketShard.js]: Right. Forgot to remove debug variables >.>

**Comment by Deivu** [comment]
Just leaving my feedback on this PR: Been running this for 2 days now and it seems to be stable enough

**Comment by xKesvaL** [comment]
I didn't saw that i received a mail for this, sorry.

I added your package, and it work perfectly.

It is fixed on the main discordjs repo ? I'm using TypeScript and typings
are not the same for this repo i guess, it gaves me a lot of errors for
types, but still work.

Thanks anyway.

**Pull request merged**